components/core/tests/test-end_to_end.cpp

```diff
@@ -0,0 +1,127 @@
+}
+
+auto get_test_input_local_path() -> std::string {
+    std::filesystem::path const current_file_path{__FILE__};
+    auto const tests_dir{current_file_path.parent_path()};
+    return (tests_dir / get_test_input_path_relative_to_tests_dir()).string();
+}
+}  // namespace
+
+// NOLINTNEXTLINE(readability-function-cognitive-complexity)
+TEST_CASE("clp-s_compression_and_extraction_no_floats", "[clp-s][end-to-end]") {
+    auto const default_target_encoded_size = 8ULL * 1024 * 1024 * 1024;  // 8 GiB
+    auto const default_max_document_size = 512ULL * 1024 * 1024;  // 512 MiB
+    auto const default_min_table_size = 1ULL * 1024 * 1024;  // 1 MiB
+    auto const default_compression_level = 3;
+    auto const default_print_archive_stats = false;
+    auto const default_ordered = false;
+    auto const default_target_ordered_chunk_size = 0;
+    auto structurize_arrays = GENERATE(true, false);
+
+    Cleanup const test_cleanup;
+
+    std::filesystem::create_directory(cTestEndToEndArchiveDirectory);
+    REQUIRE(std::filesystem::is_directory(cTestEndToEndArchiveDirectory));
+
+    clp_s::JsonParserOption parser_option{};
+    parser_option.file_paths.push_back(get_test_input_local_path());
+    parser_option.archives_dir = cTestEndToEndArchiveDirectory;
+    parser_option.target_encoded_size = default_target_encoded_size;
+    parser_option.max_document_size = default_max_document_size;
+    parser_option.min_table_size = default_min_table_size;
+    parser_option.compression_level = default_compression_level;
+    parser_option.print_archive_stats = default_print_archive_stats;
+    parser_option.structurize_arrays = structurize_arrays;
+
+    clp_s::JsonParser parser{parser_option};
+    REQUIRE(parser.parse());
+    parser.store();
+
+    REQUIRE(false == std::filesystem::is_empty(cTestEndToEndArchiveDirectory));
+
+    std::filesystem::create_directory(cTestEndToEndOutputDirectory);
+    REQUIRE(std::filesystem::is_directory(cTestEndToEndOutputDirectory));
+
+    clp_s::JsonConstructorOption constructor_option{};
+    constructor_option.archives_dir = parser_option.archives_dir;
+    constructor_option.output_dir = cTestEndToEndOutputDirectory;
+    constructor_option.ordered = false;
+    constructor_option.target_ordered_chunk_size = 0;
+    for (auto const& entry : std::filesystem::directory_iterator(constructor_option.archives_dir)) {
+        if (false == entry.is_directory()) {
+            // Skip non-directories
+            continue;
+        }
+
+        constructor_option.archive_id = entry.path().filename();
+        clp_s::JsonConstructor constructor{constructor_option};
+        constructor.store();
+    }
+
+    std::filesystem::path extracted_json_path{cTestEndToEndOutputDirectory};
+    extracted_json_path /= "original";
+    REQUIRE(std::filesystem::exists(extracted_json_path));
+
+    int result = std::system("command -v jq >/dev/null 2>&1");
```

Comment:
```suggestion
    int result{std::system("command -v jq >/dev/null 2>&1")};
```

Forgot this in the last review.